src/Features/Core/Portable/Completion/PatternMatchHelper.cs

```diff
@@ -0,0 +1,245 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using Microsoft.CodeAnalysis.PatternMatching;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    internal sealed class PatternMatchHelper : IDisposable
```

Comment:
New helper type to handle PatternMatcher related functionality, extracted from CompletionHelper. Not thread safe and must be disposed after use

Reply:
doc the threadsafety/dispose part on the type itself :)